### PR TITLE
interactive update operator warning

### PR DESCRIPTION
**Pull request by jasmingacic**
# Summary of this proposal

[sc-93553](https://app.shortcut.com/chronosphere/story/93553/calyptia-cli-update-process)

When updating with outdated CLI following Warning message will be printed:

```
# calyptia update operator --kube-namespace calyptia 
Warning: Current version 2.12.0 is less than the latest version 2.12.1
To ensure that the operator functions as intended please install latest version of the CLI, and run "calyptia update operator"
Do you want to proceed with update? (y/N)
n
Error: Aborted
exit status 1
```

Otherwise
```
# calyptia update operator --kube-namespace calyptia 
customresourcedefinition.apiextensions.k8s.io/ingestchecks.core.calyptia.com unchanged
customresourcedefinition.apiextensions.k8s.io/pipelines.core.calyptia.com unchanged
serviceaccount/calyptia-core-controller-manager unchanged
clusterrole.rbac.authorization.k8s.io/calyptia-core-manager-role configured
clusterrole.rbac.authorization.k8s.io/calyptia-core-metrics-reader unchanged
clusterrole.rbac.authorization.k8s.io/calyptia-core-pod-role unchanged
clusterrole.rbac.authorization.k8s.io/calyptia-core-proxy-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/calyptia-core-manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/calyptia-core-proxy-rolebinding unchanged
service/calyptia-core-controller-manager-metrics-service unchanged
deployment.apps/calyptia-core-controller-manager configured
Core operator manager successfully updated to version v2.12.1
```

Update can be forced by passing parameter `--force -f` to the command

### CODE REVIEW

**Review by patrick-stephens — COMMENTED** [review]
We should have an easy test for this locally right?

Also, probably a product call but it would be nice to be able to pin versions and not get this warning as well as overriding the index location for those who won't have internet access. I think we have an option to not check right so that would do that?

**Comment by jasmingacic** [comment]
> We should have an easy test for this locally right?
> 
> Also, probably a product call but it would be nice to be able to pin versions and not get this warning as well as overriding the index location for those who won't have internet access. I think we have an option to not check right so that would do that?

I can't test unless it is merged into main or released, but we do need a test for that.

Also maybe we should put all notes  into shortcut so Product can chime in

**Review by pwhelan — APPROVED** [review]
LGTM

File: cmd/operator/update.go [hunk 1]
```diff
@@ -43,6 +45,33 @@ func NewCmdUpdate() *cobra.Command {
...
 		PreRunE: func(cmd *cobra.Command, args []string) error {
+
+			operatorIndex, err := index.NewOperator()
+			if err != nil {
+				return err
+			}
+
+			if !force {
+				laststr, _ := operatorIndex.Last(cmd.Context())
+				last, _ := semver.NewSemver(laststr)
```
Comment (nicolasparada) [inline cmd/operator/update.go:56]: Please never ignore errors.

File: cmd/operator/update.go [hunk 2]
```diff
@@ -43,6 +45,33 @@ func NewCmdUpdate() *cobra.Command {
...
+				current, _ := semver.NewSemver(utils.DefaultCoreOperatorDockerImageTag)
```
Comment (nicolasparada) [inline cmd/operator/update.go:57]: Please never ignore errors.

File: cmd/operator/update.go [hunk 3]
```diff
@@ -43,6 +45,33 @@ func NewCmdUpdate() *cobra.Command {
 		Aliases: []string{"opr"},
 		Short:   "Update core operator",
 		PreRunE: func(cmd *cobra.Command, args []string) error {
+
```
Comment (nicolasparada) [inline cmd/operator/update.go:48]:
```suggestion
```

**Review by nicolasparada — CHANGES_REQUESTED** [review]
Handle errors.

**Comment by jasmingacic** [comment]
closing it in favour of future UX overhaul

**Pull request closed without merging**